testsuites/CBLTester/CBL_Functional_tests/TestSetup_FunctionalTests/test_replication_collections.py

```diff
@@ -0,0 +1,131 @@
+from keywords.SyncGateway import sync_gateway_config_path_for_mode
+from test_replication import verify_sgDocIds_cblDocIds
+
+bucket = "data-bucket"
+
+
+@pytest.fixture
+def teardown_doc_fixture():
+    def _delete_doc_if_exist(sg_client, url, db, doc_id, auth):
+        if sg_client.does_doc_exist(url, db, doc_id) is True:
+            sg_client.delete_doc(url, db, doc_id, auth=auth)
+    yield _delete_doc_if_exist
+
+
+@pytest.fixture(scope="function")
+def scope_collection_test_fixture(params_from_base_test_setup):
+    random_suffix = str(uuid.uuid4())[:8]
+    db_prefix = "db_"
+    scope_prefix = "scope_"
+    collection_prefix = "collection_"
+    cbl_db_name = db_prefix + random_suffix
+    scope = scope_prefix + random_suffix
+    collection = collection_prefix + random_suffix
+    need_sgw_admin_auth = params_from_base_test_setup["need_sgw_admin_auth"]
+    sg_username = "scope_collection_user" + random_suffix
+    log_info("****")
+    log_info(sg_username)
+    sg_password = "password"
+    data = {"bucket": bucket, "scopes": {scope: {"collections": {collection: {}}}}, "num_index_replicas": 0}
+    sg_db = "db"
+
+    sg_client = sg_url = sg_admin_url = None
+    cluster_config = params_from_base_test_setup["cluster_config"]
+    mode = params_from_base_test_setup["mode"]
+    sg_config = sync_gateway_config_path_for_mode("sync_gateway_default_functional_tests", mode)
+    sg_admin_url = params_from_base_test_setup["sg_admin_url"]
+    sg_url = params_from_base_test_setup["sg_url"]
+    base_url = params_from_base_test_setup["base_url"]
+    sg_blip_url = params_from_base_test_setup["target_url"]
+
+    c = cluster.Cluster(config=cluster_config)
+    c.reset(sg_config_path=sg_config)
+    auth = need_sgw_admin_auth and [RBAC_FULL_ADMIN['user'], RBAC_FULL_ADMIN['pwd']] or None
+    admin_client = Admin(c.sync_gateways[0])
+    db = Database(base_url)
+    col_obj = Collection(base_url)
+    doc_obj = Document(base_url)
+    sg_client = MobileRestClient()
+    db_config = db.configure()
+
+    sync_gateway_version = params_from_base_test_setup["sync_gateway_version"]
+
+    if sync_gateway_version < "2.0.0":
+        pytest.skip('This test cannot run with sg version below 2.0')
+
+    cluster_helper = ClusterKeywords(cluster_config)
+    topology = cluster_helper.get_cluster_topology(cluster_config)
+    cbs_url = topology["couchbase_servers"][0]
+    cb_server = couchbaseserver.CouchbaseServer(cbs_url)
+    cb_server.get_bucket_names()
+    does_scope_exist = cb_server.does_scope_exist(bucket, scope)
+    if does_scope_exist is False:
+        cb_server.create_scope(bucket, scope)
+        cb_server.create_collection(bucket, scope, collection)
+
+    # sgw database creation
+    if admin_client.does_db_exist(sg_db) is True:
+        admin_client.delete_db(sg_db)
+        admin_client.create_db(sg_db, data)
+    else:
+        admin_client.create_db(sg_db, data)
+    log_info("sgw db created")
```

Comment:
Do we need these prints?

Reply:
No, this is an indication. Will rectify this.